alcs-frontend/src/app/features/notice-of-intent/overview/overview.component.ts

```diff
@@ -108,4 +110,34 @@ export class OverviewComponent implements OnInit, OnDestroy {
         });
     }
   }
+
+  onTogglePortalVisible() {
+    if (this.isHiddenFromPortal) {
+      this.confirmationDialogService
+        .openDialog({
+          body: 'Are you sure you want to remove the access restriction for the L/FNG and public? If you continue, this File ID could return search results for the L/FNG and the public. Standard rules for showing/hiding content will apply.',
+        })
+        .subscribe((didConfirm) => {
+          this.isHiddenFromPortal = didConfirm;
+          if (didConfirm && this.noticeOfIntent) {
+            this.noticeOfIntentDetailService.update(this.noticeOfIntent.fileNumber, {
+              hideFromPortal: true,
+            });
+          }
+        });
+    } else {
+      this.confirmationDialogService
+        .openDialog({
+          body: 'If you continue, this File ID will not return any search results for the L/FNG and the public. Please add a journal note to explain why this file is restricted.',
+        })
+        .subscribe((didConfirm) => {
+          this.isHiddenFromPortal = !didConfirm;
+          if (didConfirm && this.noticeOfIntent) {
+            this.noticeOfIntentDetailService.update(this.noticeOfIntent.fileNumber, {
+              hideFromPortal: false,
+            });
+          }
+        });
+    }
+  }
 }
```

Comment:
How come we're updating to `hideFromPortal: true` when we want to remove the restriction, and it is the opposite when we want to add the restriction? Wouldn't we want it to be `hideFromPortal` to be false when we want to be able to view it?

Reply:
Ah good catch! I think I got the messages backwards but the logic is correct.

Reply:
Updated! The logic might look backwards in the if/else, but the value changes immediately before the onChange is fired.

Reply:
Makes sense! Application looks good now, but NOI still needs update

Reply:
NOI updated